addons/print/models/print_printer.py

```diff
@@ -97,28 +97,45 @@ def _check_groups(self):
                                       printer.name)
 
     @api.multi
-    def printers(self, raise_if_not_found=False):
+    def printers(self, report_type=None, raise_if_not_found=False):
         """Determine printers to use"""
+        printers = self.browse()
 
-        # Start with explicitly specified list of printers, falling
-        # back to user's default printer, falling back to system
-        # default printer
-        printers = (self or self.env.user.printer_id or
-                    self.search([('is_default', '=', True),
-                                 ('group_id', '=', False)]))
+        if self and (report_type is None or self.report_type == report_type):
+            printers = self
+        else:
+            printers = self.env.user._get_printer(report_type)
+
+        if not self and not printers:
+            # Fall back to system default
+            system_default_args = [("is_default", "=", True), ("group_id", "=", False)]
+            if report_type is not None:
+                system_default_args.append(("report_type", "=", report_type))
+            printers = self.search(system_default_args)
 
         # Iteratively reduce any printer groups to their user or
         # system default printers
-        while printers.filtered(lambda x: x.is_group):
-            printers = printers.mapped(lambda p: (
-                p if not p.is_group else
-                ((p.child_ids & self.env.user.printer_ids) or
-                 (p.child_ids.filtered(lambda x: x.is_default)))
-            ))
+        while printers.filtered(lambda p: p.is_group):
+            printers = printers.mapped(
+                lambda p: (
+                    p
+                    if not p.is_group and report_type is not None and p.report_type == report_type
```

Comment:
Is this supposed to be `and (report_type is None or p.report_type == report_type)` ?

Reply:
Yeah good spot, will that.

Basically I only want to check `report_type` on `p` if `report_type` is specified in the call to `printers`.

Reply:
Have now pushed a fix to change this.